pkg/operator/controller/ingress/controller.go

```diff
@@ -185,9 +186,12 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 
 	// The ingresscontroller is safe to process, so ensure it.
 	if err := r.ensureIngressController(ingress, dnsConfig, infraConfig, ingressConfig, apiConfig); err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to ensure ingresscontroller: %v", err)
+		if retryable, isRetryable := err.(retryable.Error); isRetryable {
+			log.Error(err, "got retryable error; requeueing", "after", retryable.After())
+			return reconcile.Result{RequeueAfter: retryable.After()}, nil
+		}
+		return reconcile.Result{}, err
```

Comment:
Bikeshed... go type switch is another style alternative:

```go
switch e := err.(type) {
  case retryable.Error:
    log.Error(e, "got retryable error; requeueing", "after", e.After())
    return reconcile.Result{RequeueAfter: e.After()}, nil
  default:
    return reconcile.Result{}, err
}
```

Reply:
That works for me.  I'll update it in the next push.

Reply:
Done.